### DISCUSSION TITLE
All roads lead to Rome : user to developer, encouraging developers of the second tier

### DISCUSSION THREAD

**Discussion by sriramab** [gama-platform/gama#62] [General]
This is roughly the same idea that is currently addressed by importing a gaml file to access reusable objects/functions/logic from source external to the model. In its current form, it is always an open-source. But this enhancement proposes to make "it" a closed source with an option to open it when necessary. This  enchancement has the potential to open up opportunities similar to **App Store**, **Play Store** , **Marketplace** of user-generated plugins. 

To extend  and reinforce the general idea making GAMA more accessible and inviting...... the idea of exchanging and distributing models or its components:
 - to increase their reuse and reduce the model building time  
 - to encourage normal users to develop plugins/ add-ins from their reusable components in their code.
 - to reduce "re-inventing the wheel" hiccups

     a feasible proposal appears to be to the ability to "translate" GAML code into a some form of a package (closed source by default) and one  that can be imported as a plugin/ add-in along with some form of a manifest that describes what is exposed (available) to the users from this closed source. 

The general idea is to encourage non-developers to package their great ideas (GAML code, GAML code chunks in form of models, operators, actions)  in a shareable form for the downstream modelers to build onto the previous work. Some examples could be packaging of actions, aspects, visualizations, color schemes, wrapper functions that combine a set of operators to achieve common tasks etc. 

TL;DR

`The phrase "standing on the shoulders of giants" is a metaphor which means "using the understanding gained by major thinkers who have gone before in order to make intellectual progress".It is a metaphor of dwarfs standing on the shoulders of giants (Latin: nanos gigantum humeris insidentes) and expresses the meaning of "discovering truth by building on previous discoveries".`
